services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/BandwidthSpec.groovy

```diff
@@ -0,0 +1,73 @@
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+import org.openkilda.testing.service.northbound.NorthboundService
+import org.openkilda.testing.service.topology.TopologyEngineService
+import org.springframework.beans.factory.annotation.Autowired
+
+class BandwidthSpec extends BaseSpecification {
+
+    @Autowired
+    TopologyDefinition topology
+    @Autowired
+    TopologyEngineService topologyEngineService
+    @Autowired
+    FlowHelper flowHelper
+    @Autowired
+    NorthboundService northboundService
+
+    def "Available bandwidth on ISLs changes respectively when creating a flow"() {
+        when: "Create flow with a valid bandwidth"
+        def maximumBandwidth = 1000
+        List<List<PathNode>> allPaths = []
+
+        def switches = topology.getActiveSwitches()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            allPaths = topologyEngineService.getPaths(src.dpId, dst.dpId)*.path
+            allPaths.size() > 1
+        }
+        srcSwitch && dstSwitch
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        def links = northboundService.getAllLinks()
+        def forwardLink = links.find() {
+            it.path.first().switchId == flow.source.datapath &&
+                    it.path.last().switchId == flow.destination.datapath
+        }
+        def reverseLink = links.find {
+            it.path.first().switchId == flow.destination.datapath &&
+                    it.path.last().switchId == flow.source.datapath
+        }
+        def availableBandwidthForward = forwardLink.availableBandwidth
+        def availableBandwidthReverse = reverseLink.availableBandwidth
+
+        flow.maximumBandwidth = maximumBandwidth
+        northboundService.addFlow(flow)
+
+        then: "Flow is really created and has 'Up' status"
+        northboundService.getFlowStatus(flow.id).status == FlowState.UP
+
+        and: "Available bandwidth on ISL is changed in accordance with flow maximum bandwidth"
+        def actualLinks = northboundService.getAllLinks()
+        def actualForwardLink = actualLinks.find {
+            it.path.first().switchId == flow.source.datapath &&
+                    it.path.last().switchId == flow.destination.datapath
+        }
+        def actualReverseLink = actualLinks.find {
+            it.path.first().switchId == flow.destination.datapath &&
+                    it.path.last().switchId == flow.source.datapath
+        }
+
+        actualForwardLink.availableBandwidth == availableBandwidthForward - maximumBandwidth
+        actualReverseLink.availableBandwidth == availableBandwidthReverse - maximumBandwidth
+        
+        cleanup: "Delete created flow"
+        flow?.id && northboundService.deleteFlow(flow.id)
+    }
```

Comment:
Please look into "Should be able to reroute to a better path if it has enough bandwidth" where I check the availableBandwidth after reroute. Your case should perform pretty much the same logic but for Create/Update operations (not reroute). What I recommend to do:
- Pick up 2 switches (it actually does not matter which switches to pick, so probably just pick 2 first switches or 2 last switches or whatever, but not random to ensure repeatability)
- get all available paths for these switches by calling `topologyEngineService.getPaths`
- create a method in PathHelper that will take a path as input and will return the actual **cost** of that path
- given you have such tool you can now find a path that will have the lowest cost (Kilda will select this path when building a flow)
- Given you have potential path that will be used, you can now call `def involvedIsls = pathHelper.getInvolvedIsls(myPath)` to get a list of ISLs involved in your path.
- ^ This will be a list of our **internal ISL representations**(from topology.yaml). In order to get actual ISL state (from NB) for each of these ISLs you'll have to call `islUtils.getIslInfo(myInternalIslRepresentation)`. So that it becomes:
```
def allLinks = northbound.getAllLinks()
def actualLinksUsedInOurPath = involvedIsls.collect { islUtils.getIslInfo(allLinks, it) }
```
- you can see the current bandwidth of all these isls: `println actualLinksUsedInOurPath*.availableBandwidth`
^this is you current state before building a flow
--------------------------------------
- Build a flow for your selected switches
- Call `PathHelper.convert(northboundService.getFlowPath(flow.id))` to get the Path which was chosen for your flow by Kilda
- you can compare now that the chosen path matches the path you have predicted above (matches the path with lowest cost)
- now rerun the getAllLinks code and recheck the updated availableBandwidth matches your expectations